### PR TITLE
add lacp_rate to portchannel

### PR DESCRIPTION
**Pull request by globaltrouble**
#### What I did

Make lacp_rate configurable for portchannel.

```
	Option specifying the rate in which we'll ask our link partner
	to transmit LACPDU packets in 802.3ad mode.  Possible values
	are:

	slow
		Request partner to transmit LACPDUs every 30 seconds

	fast
		Request partner to transmit LACPDUs every 1 second

	The default is slow.
```

#### Why I did it

In case of slow lacp_rate configuration link down will be detected in 60-90 seconds, it may be to long (for example for MCLAG high availability), in case of using ` --fast-rate=true` link down will be detected in 2-3 seconds.

#### How I did it
   
* add optional argument to `config portchannel` command, default=slow for backward compatibility. (this PR)
* parse argument in `teammgr` and forward it to `teamd` (other PR: https://github.com/Azure/sonic-swss/pull/2121)
* update docs https://github.com/Azure/SONiC/pull/937

#### How to verify it

Confgiure bond on other side, then configure portchannel and sniff the traffic from it.

```
config portchannel add PortChannel0001 --fast-rate=true
config portchannel member add PortChannel0001 Ethernet0
config interface ip add PortChannel0001  192.168.1.2/24
tcpdump -ne
```

### CODE REVIEW

**Comment by novikauanton** [comment]
@stcheng could you please review it?

**Comment by msosyak** [comment]
Related PR's:
- https://github.com/Azure/SONiC/pull/937
- https://github.com/Azure/sonic-swss/pull/2121

**Comment by msosyak** [comment]
@qiluo-msft  @bingwang-ms  Help to merge this, Please.

File: config/main.py [hunk 1]
```diff
@@ -1720,8 +1720,11 @@ def portchannel(ctx, namespace):
 @click.argument('portchannel_name', metavar='<portchannel_name>', required=True)
 @click.option('--min-links', default=1, type=click.IntRange(1,1024))
 @click.option('--fallback', default='false')
+@click.option('--fast-rate', default='false',
+              type=click.Choice(['true', 'false'],
+                                case_sensitive=False))
 @click.pass_context
-def add_portchannel(ctx, portchannel_name, min_links, fallback):
+def add_portchannel(ctx, portchannel_name, min_links, fallback, fast_rate):
```
Comment (nazariig) [inline config/main.py:1727]: @globaltrouble please add a UTs to cover this case

**Review by nazariig — CHANGES_REQUESTED** [review]

**Comment by bratashX** [comment]
will be good to include these changes into 202111 branch

**Comment by qiluo-msft** [comment]
Please solve the conflicts

File: tests/portchannel_test.py [hunk 2]
```diff
@@ -60,7 +60,30 @@ def test_delete_non_existing_portchannel(self):
...
         assert "Error: PortChannel0005 is not present." in result.output
-        
+
+    def test_add_portchannel_with_fast_rate(self):
+        runner = CliRunner()
+        db = Db()
+        obj = {'db':db.cfgdb}
+
+        # add a portchannel with fats rate
+        result = runner.invoke(config.config.commands["portchannel"].commands["add"], ["PortChannel0005", "--fast-rate", "True"], obj=obj)
```
Comment (nazariig) [inline tests/portchannel_test.py:70]: @globaltrouble please remove all created objects on teardown. Also suggest to have parametrization for `True/False` values

Comment (msosyak) [reply tests/portchannel_test.py]: @nazariig Parametrization added. Regarding "remove all created objects on teardown" why it is needed? The DB is instantiated for each test from the template  https://github.com/Azure/sonic-utilities/blob/3600639c16079f17089efd655c818022987032bb/tests/mock_tables/config_db.json

File: config/main.py [hunk 3]
```diff
@@ -1946,8 +1946,11 @@ def portchannel(db, ctx, namespace):
 @click.argument('portchannel_name', metavar='<portchannel_name>', required=True)
 @click.option('--min-links', default=1, type=click.IntRange(1,1024))
 @click.option('--fallback', default='false')
+@click.option('--fast-rate', default='false',
+              type=click.Choice(['true', 'false'],
```
Comment (nazariig) [inline config/main.py:1950]: @globaltrouble IMHO, better to use flags or `click.BOOL`

Comment (msosyak) [reply config/main.py:1950]: In the current implementation, the syntax is the same for `fallback` and for `fast-rate`. Booth requires an argument. If change type to bool or flags then I think it should be done for both options together is separate PR

**Comment by msosyak** [comment]
@nazariig Do you have other comments?

**Comment by msosyak** [comment]
@nazariig @prsunny could we merge this?

**Comment by msosyak** [comment]
@nazariig Could you approve, please?

**Review by nazariig — APPROVED** [review]

**Review by judyjoseph — APPROVED** [review]
LGTM. Thinking from db migration angle as this is a new field being added -- I guess is ok as the default is false.

File: doc/Command-Reference.md [hunk 4]
```diff
@@ -6925,12 +6925,13 @@ When any port is already member of any other portchannel and if user tries to ad
 Command takes two optional arguements given below.
 1) min-links  - minimum number of links required to bring up the portchannel
 2) fallback - true/false. LACP fallback feature can be enabled / disabled.  When it is set to true, only one member port will be selected as active per portchannel during fallback mode. Refer https://github.com/Azure/SONiC/blob/master/doc/lag/LACP%20Fallback%20Feature%20for%20SONiC_v0.5.md for more details about fallback feature.
+3) fast-rate - true/false, default is flase (slow). Option specifying the rate in which we'll ask our link partner to transmit LACPDU packets in 802.3ad mode. slow - request partner to transmit LACPDUs every 30 seconds, fast - request partner to transmit LACPDUs every 1 second. In slow mode 60-90 seconds needed to detect linkdown, in fast mode only 2-3 seconds.
```
[the remark this answers is not in the document]
Comment (msosyak) [reply doc/Command-Reference.md]: Fixed

**Comment by msosyak** [comment]
@judyjoseph Could you help to merge this?

**Comment by msosyak** [comment]
@qiluo-msft Please, help to merge.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**